src/main/java/seedu/duke/command/EditTransactionCommand.java

```diff
@@ -0,0 +1,219 @@
+                isSuccess = !originalAmount.equals(newAmount);
+            } else if (getArg(CATEGORY_ARG) != null && !getArg(CATEGORY_ARG).isBlank()) {
+                isSuccess = handleCategoryEdit(expense, idx);
+            }
+        }
+        if (!isSuccess) {
+            throw new DukeException(ERROR_MSG);
+        }
+        printSuccess(ui, transactionDescription, idx + 1); // idx + 1 for format to show to user
+    }
+
+    private int getTransactionMaxSize(String type) {
+        int maxSize = 0;
+        if (type.equals("in")) {
+            maxSize = StateManager.getStateManager().getIncomesSize();
+        } else if (type.equals("out")) {
+            maxSize = StateManager.getStateManager().getExpensesSize();
+        }
+        return maxSize;
+    }
+
+    private int parseIdx(int maxSize) throws DukeException {
+        int index = Integer.parseInt(getDescription());
+        if (index < 1 || index > maxSize) {
+            throw new DukeException(INVALID_IDX);
+        }
+        return index;
+    }
+    
+    private void printSuccess(Ui ui, String description, int idx) {
+        String type = getArg("type").toLowerCase();
+        String transactionType = type.equals("in") ? "income" : "expense";
+        String msg = "Successfully edited " + transactionType + " no." + idx + " " + description;
+        ui.print(msg);
+    }
+
+    private boolean handleGoalEdit(Income income, int idx) throws DukeException {
+        boolean isSuccess = false;
+        boolean foundNewGoal = false;
+        Goal originalGoal = income.getGoal();
+        String newGoalDescription = getArg(GOAL_ARG);
+        if (originalGoal.getDescription().equals(newGoalDescription)) {
+            throw new DukeException(SAME_GOAL);
+        }
+
+        for (Goal goal : StateManager.getStateManager().getAllGoals()) {
+            if (goal.getDescription().equals(newGoalDescription)) {
+                StateManager.getStateManager().getIncome(idx)
+                        .setGoal(goal);
+                isSuccess = !originalGoal.equals(income.getGoal());
+                foundNewGoal = true;
+                break;
+            }
+        }
+        if (!foundNewGoal) {
+            throw new DukeException("Please add " + newGoalDescription + " as a goal first.");
+        }
+        return isSuccess;
+    }
+
+    private boolean handleCategoryEdit(Expense expense, int idx) throws DukeException {
+        boolean isSuccess = false;
+        boolean foundNewCategory = false;
+        Category originalCategory = expense.getCategory();
+        String newCategoryDescription = getArg(CATEGORY_ARG);
+        if (originalCategory.getName().equals(newCategoryDescription)) {
+            throw new DukeException(SAME_CATEGORY);
+        }
+
+        for (Category category : StateManager.getStateManager().getAllCategories()) {
+            if (category.getName().equals(newCategoryDescription)) {
```

Comment:
getCategoryIndex from StateManager might be helpful here, returns -1 if not found.